HARK/ConsumptionSaving/ConsRiskyContribModel.py

```diff
@@ -0,0 +1,1768 @@
+"""
+This file contains classes and functions for representing, solving, and simulating
+agents who must allocate their resources among consumption, saving in a risk-free
+asset (with a low return), and saving in a risky asset (with higher average return).
+"""
+import numpy as np
+from copy import deepcopy
+from HARK import MetricObject, NullFunc  # Basic HARK features
+from HARK.ConsumptionSaving.ConsIndShockModel import (
+    utility,  # CRRA utility function
+    utility_inv,  # Inverse CRRA utility function
+    utilityP,  # CRRA marginal utility function
+    utilityP_inv,  # Inverse CRRA marginal utility function
+    init_lifecycle,
+)
+
+from HARK.ConsumptionSaving.ConsRiskyAssetModel import (
+    RiskyAssetConsumerType,
+    risky_asset_parms,
+    init_risky_asset,
+)
+
+from HARK.distribution import calc_expectation
+
+from HARK.interpolation import (
+    LinearInterp,  # Piecewise linear interpolation
+    BilinearInterp,  # 2D interpolator
+    TrilinearInterp,  # 3D interpolator
+    ConstantFunction,  # Interpolator-like class that returns constant value
+    IdentityFunction,  # Interpolator-like class that returns one of its arguments
+    ValueFuncCRRA,
+    MargValueFuncCRRA,
+    DiscreteInterp,
+)
+
+from HARK.utilities import make_grid_exp_mult
+
+
+class RiskyContribConsumerType(RiskyAssetConsumerType):
+    """
+    A consumer type with idiosyncratic shocks to permanent and transitory income,
+    who can save in both a risk-free and a risky asset but faces frictions to
+    moving funds between them. The agent can only consume out of his risk-free
+    asset.
+    
+    The frictions are:
+        - A proportional tax on funds moved from the risky to the risk-free
+         asset.
+        - A stochastic inability to move funds between his accounts.
+    
+    To partially avoid the second friction, the agent can commit to have a
+    fraction of his labor income, which is usually deposited in his risk-free
+    account, diverted to his risky account. He can change this fraction
+    only in periods where he is able to move funds between accounts.
+    """
+
+    time_inv_ = deepcopy(RiskyAssetConsumerType.time_inv_)
+    time_inv_ = time_inv_ + ["DiscreteShareBool"]
+
+    state_vars = RiskyAssetConsumerType.state_vars + [
+        "nNrm",
+        "mNrmTilde",
```

Comment:
maybe would be good to document what these new variables mean in a docstring.

In particular I'm not sure how to interpret the Tilde here.

Reply:
Addressed in https://github.com/econ-ark/HARK/pull/832/commits/efa3cd7d50a2585b201b127e4179aca5999e33e8